packages/langium/src/test/langium-test.ts

```diff
@@ -101,33 +101,43 @@ export function expectFoldings(services: LangiumServices): (input: ExpectedBase)
     };
 }
 
-function textDocumentParams(document: LangiumDocument): { textDocument: TextDocumentIdentifier } {
+export function textDocumentParams(document: LangiumDocument): { textDocument: TextDocumentIdentifier } {
     return { textDocument: { uri: document.textDocument.uri } };
 }
 
+export interface AssertCompletion extends ExpectedBase {
+    index: number
+    assertCompletionList: (completions: CompletionList) => void;
+}
+
 export interface ExpectedCompletion extends ExpectedBase {
     index: number
     expectedItems: Array<string | CompletionItem>
 }
 
-export function expectCompletion(services: LangiumServices): (completion: ExpectedCompletion) => Promise<void> {
+export function expectCompletion(services: LangiumServices): (completion: AssertCompletion|ExpectedCompletion) => Promise<void> {
```

Comment:
Instead of accepting different kinds of assertion objects, these types should be structured more clearly:

```ts
interface ExpectedItemCompletion {
    index: number
    expectedItems: Array<string | CompletionItem>
}
interface ExpectedCallbackCompletion {
    index: number;
    assert: (completions: CompletionList) => void;
}
type ExpectedCompletion = ExpectedItemCompletion | ExpectedCallbackCompletion;
```